File: backend/src/models/post.js
import { model, Schema, Types } from "mongoose";

const postSchema = new Schema({
  title: {
    type: String,
    required: true,
  },
  tags: {
    type: String,
  },
  content: {
    type: String,
    required: true,
  },
  user: {
    type: Types.ObjectId,
    ref: "User",
  },
});

const Post = model("Post", postSchema);

export default Post;

File: frontend/lib/models/Post.dart
class Post {
  String id;
  String title;
  String content;
  String tags;
  String user;

  Post(
      {required this.id,
      required this.title,
      required this.content,
      required this.tags,
      required this.user});

  factory Post.fromJson(Map<String, dynamic> json) => Post(
        id: json['_id'],
        title: json['title'],
        content: json['content'],
        tags: json['tags'],
        user: json['user'],
      );
}
